### PR TITLE
samples: Bluetooth: df: Add missing harness=bluetooth to DF sample.yaml

### LINKED ISSUE #43552: samples: bluetooth: direction_finding: Sample fails on nrf5340
**Describe the bug**
Samples defined at https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/direction_finding_connectionless_rx/sample.yaml and https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/bluetooth/direction_finding_connectionless_tx/sample.yaml fail on nrf5340dk_nrf5340_cpuapp.:
Failing ones: sample.bluetooth.direction_finding_connectionless_rx.aod and sample.bluetooth.direction_finding_connectionless.aoa
(FYI The other 2 scenarios in those yamls are skipped since we don't have "bluetooth" harness defined.)

Probably those tests fail on other allowed platforms but I cannot verify this. 
I think there are 2 issues here.
- the tests have no means of validating a results. Neither ztest nor console harnesses are used, hence such tests will always end up with a timeout
- there are errors present in the output

**To Reproduce**
1. `scripts/twister -T samples/bluetooth/direction_finding_connectionless_rx/ -p nrf5340dk_nrf5340_cpuapp --device-testing --device-serial /dev/ttyACM2 -v -v`
2. See error

**Expected behavior**
Tests pass or are skipped

**Impact**
Not clear 

**Logs and console output**
```
DEBUG   - DEVICE: *** Booting Zephyr OS build zephyr-v3.0.0-883-g6650a5304011  ***
DEBUG   - DEVICE: Starting Connectionless Locator Demo
DEBUG   - DEVICE: Bluetooth initialization...failed (err -11)
DEBUG   - DEVICE: success
DEBUG   - DEVICE: Scan callbacks register...success.
DEBUG   - DEVICE: Periodic Advertising callbacks register...success.
DEBUG   - DEVICE: Start scanning...failed (err -11)
```
and
```
*** Booting Zephyr OS build zephyr-v3.0.0-838-gc050d1f2ecb6  ***
Starting Direction Finding periodic advertising Beacon Demo
Bluetooth initialization...failed (err -11)
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used

**Additional context**
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Some build configurations in direction finding samples
sample.yaml didn't have harness=bluetooth. That enables
these configurations to be be executed on hardware by
CI. Those runs end causing CI failures.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Review by Thalley — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by PerMac — DISMISSED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
> Some of them could be executed on hardware by
> CI. That ends with CI failures.

I am not following how this can be true, you already have a harness set which should prevent this from running in CI. If this sample is not designed to be run, not sure why we have it as a sample in the first place. Build Only is only for unrunnable test code (i.e. build tests like for drivers).

If a sample with harness: bluetooth is running in CI, that is a bug somewhere else.

**Comment by ppryga** [comment]
> I am not following how this can be true, you already have a harness set which should prevent this from running in CI.

As of now, according to @PerMac there is no harness for `bluetooth`, but that could change (not very likely but still).
Use of `build_only` is more elegant way to achieve it.

> If this sample is not designed to be run, not sure why we have it as a sample in the first place. Build Only is only for unrunnable test code (i.e. build tests like for drivers).

Direction finding samples are aimed to be `samples` for potential end users that want to get familiar with BLE direction finding feature. For nRF52 SOC they can be executed on HW but there is no meaningful output besides that they have started.
For nRF53 there is a need to run build and flash a hex file for network core to successfully run the sample.
If samples are used in pairs e.g. direction_finding_connectionless_rx and direction_finding_connectionless_tx then there is a possibility to validate the DF feature.

**Comment by PerMac** [comment]
@nashif There is a small misunderstanding. Upstream's CI won't run this samples, this is correct. Piotr is referring to our internal CI which is running twister on HW. Half of those scenarios didn't have harness defined, hence twister is trying to execute them on a hardware and ends up with a timeout. 
Like Piotr explained, those samples can be executed, but not with twister. I have doubts about usage of `harness: bluetooth`. I think right now it is used mainly to tell twister to skip the execution of such sample. IMO it is misguiding. It suggest that there is, or at least can be in future, a way to verify such test with a proper harness. The problem is there is no standard interface on the tests side based on which its result can be identified in a standardized way. Therefore, we won't be able to introduce any bluetooth harness without modification of all tests which have it selected now.  
I don't mind aligning samples from this issue with the others by adding harness: bluetooth. Is this what you suggest to do @nashif?

**Comment by PerMac** [comment]
@nashif  One of these scenarios do not have harness set and it is the one not filtered by on-target CI:
  sample.bluetooth.direction_finding_connectionless_rx.aod

**Comment by ppryga** [comment]
@nashif @PerMac I can add `harness` for every build type and remove `build_only` but in my humble opinion `build_only` is more self explanatory and explicit. 

There is another option, I'm not sure if that is possible, to use HW but for selected platforms only (no nRF53).

**Comment by nashif** [comment]
> @nashif @PerMac I can add `harness` for every build type and remove `build_only` but in my humble opinion `build_only` is more self explanatory and explicit.

build_only is supposed to be used only for something that can not be run anywhere. If this sample is not runnable or usable, then it should not be a sample in the first place.
I do not have a link to the history of this (was 5 years ago I think), but we did at some point move all build_only code to either depend on hw they need to run or added the harness as a placeholder to indicate that you need a special setup to run the sample and to prevent CI from running it.

Long story short, build_only is the wrong solution here, you should use harness.

**Comment by ppryga** [comment]
@nashif `build-only` removed and missing `harness=bluetooth` added to configurations where it was missing.

**Review by PerMac — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Review by nashif — APPROVED** [review]
